torch_xla/core/xla_model.py

```diff
@@ -533,8 +553,7 @@ def all_gather(value, dim=0, groups=None, output=None, pin_layout=True):
     A tensor which has, in the ``dim`` dimension, all the values from the
     participating replicas.
   """
-  if pin_layout and xla_device_hw(
-      value.device) in ('TPU', 'GPU', 'XPU') and output == None:
```

Comment:
I think we had it because CPU was not supported at some point. Do you need to remove it because it will break dynamo?

Reply:
Yea.